plugin-dev/Source/Sentry/Private/Apple/SentryIdApple.cpp

```diff
@@ -7,7 +7,12 @@
 
 SentryIdApple::SentryIdApple()
 {
+	// `SentryId` definition was moved to Swift so its name that can be recognized by UE should be taken from "Sentry-Swift.h" to successfully load class on Mac
+#if PLATFORM_MAC
+	IdApple = [[SENTRY_APPLE_CLASS(_TtC6Sentry8SentryId) alloc] init];
```

Comment:
I think it's related to the way we're consuming `sentry-cocoa` on Mac - there it is necessary to load each symbol from the `sentry.dylib` [manually](https://github.com/getsentry/sentry-unreal/blob/595bf09c567561e2842c74976d61dd9d195f3ea9/plugin-dev/Source/Sentry/Private/Apple/Convenience/SentryMacro.h#L13) while on iOS all of the `Sentry.framework` classes are available right out of the box.

Reply:
After investigating this a bit more it seems to be a common practice to use those mangled class names in a scenario like ours. Linking the [same thing](https://github.com/getsentry/sentry-dotnet/blob/1621b6c2828e414fcf7797b8acc5e32e91a1bc4a/src/Sentry.Bindings.Cocoa/ApiDefinitions.cs#L2437) in .NET SDK here just in case.